### DISCUSSION TITLE
Specific notes act differently when legato is enabled

### DISCUSSION THREAD

**Discussion by audio2000** [FluidSynth/fluidsynth#1412] [General]
I apologize if this has already been corrected, it seems that many Linux plugins that use Fluidsynth do not seem to keep up with updates of Fluidsynth itself.
If not, I may have found a possible bug if anyone is interested.
I also apologize because some of this may be difficult to explain well.

`ISSUE TOPIC`
When legato is enabled I noticed that two specific note overlaps that are notes side by side, seem to generate an almost portamento style effect compared to all of the other side by side legato notes tested.

`HOW TO REPRODUCE`
Legato is achieved when user holds down a note and then overlaps playing with another note.  
The two specific note overlaps I notice this issue with are....
-when you hold down D and play the next note E
-when you hold down B and play the next note C
when you do this you almost hear more of a portamento kind of effect occurring, the notes are not as defined sounding as when you do this with other notes.
My tests were most often done using lower octaves of these notes D1 D2 D3 but I think I remember this occurring with the higher octaves of these notes as well.

To really hear this portamento kind of sound I am speaking of, you can hold down the first note (for example D) and then keep tapping and release next note (for example E).

This occurred with more than one soundfont so it does not appear to be instrument specific.

Again the unfortunate thing is that I cannot test newer versions of Fluidsynth at the moment.  On top of that even if this is a bug and corrected as mentioned it does not seem like plugins update to newer versions, but figured if it is a bug I should mention what I have found, maybe could help others who have access to newer versions of Fluidsynth.

If anyone willing to try and verify if they could generate the same results, please mention it here.
If this is a bug if someone could move this over to the Issues section, or let me know and I can copy and paste it over if needed.

Thank You

**Comment by derselbst** [comment]
I am not using legato, portament, or poly / mono. But there is quite a lot documentation available here: https://github.com/FluidSynth/fluidsynth/tree/master/doc/polymono

And I'm pinging @jjceresa  since he implemented it and might be able to provide a direct answer.

**Comment by audio2000** [comment]
Thank you for your reply.
Does not seem like that developer has been around for quite a few years now so does not look like he will be responding.
Not sure if you would be able to answer this, but do you by any chance know if those .txt files mentioned in the link you provided are the actual code used by Fluidsynth to make use of things like legato?  Or are these .txt files more just informational and there is different actual code responsible for things like legato.

Thank you again for at least trying to contact that developer for me.

**Reply by derselbst** [reply]
The txt files are [config files for fluidsynth's shell](https://github.com/FluidSynth/fluidsynth/wiki/FluidSynth-configuration-files#example-configuration-file), this is not related to the actual implementation. The main part of the implementation is included here:
https://github.com/FluidSynth/fluidsynth/blob/master/src/synth/fluid_synth_monopoly.c

**Comment by jjceresa** [comment]
@audio2000

>ISSUE TOPIC
>When legato is enabled I noticed ....

Sorry, poly/mono works fine here, as described by  FluidPolyMono-0004.pdf chapter 3.
So none of the issue you describe above can be reproduced.

>Not sure if you would be able to answer this, but do you by any chance know if those .txt files mentioned in
>the link you provided are the actual code used by Fluidsynth to make use of things like legato?

Yes, those .txt files are usable directly as described by readme.txt FluidPolyMono-0004.pdf chapter 3.1.
You need the fluidsynth console application. https://github.com/FluidSynth/fluidsynth/wiki/UserManual#shell-commands

I recommend using this tool while using any MIDI keyboard connected to fluidsynth. Option -d and -v  will allow to be aware of MIDI messages send over your shoulder and received by fluidsynth.

**Comment by audio2000** [comment]
@jjceresa 
Oh I am sorry I have not seen your name around in a while, I was not expecting a response.
Thank you for testing this out, strange that I was getting these results, might have just been the instruments I was testing at the time.
I will look into the suggestions that you have mentioned.
derselbst mentioned that you were responsible for implementing this into Fluidsynth, so if anything at least this gives me the opportunity to thank you so much for adding this, really great feature, really appreciated.
Thanks again for your response and for adding this.

derselbst
If you should read this, thank you again for making the contact and for the other information that you have provided.
